### PR TITLE
Convert to IPBlock

### PR DESCRIPTION
**Pull request by YairSlobodin1**

### CODE REVIEW

File: pkg/ir/spec.go [hunk 1]
```diff
@@ -44,7 +47,7 @@ type (
 		Name string
 
 		// list of CIDR / Ip addresses.
-		Values []IP
+		Values []*ipblock.IPBlock
```
Comment (zivnevo) [inline pkg/ir/spec.go:50]: Consider renaming `Values` to `CIDRs` or `IPAddrs`

File: pkg/ir/spec.go [hunk 2]
```diff
@@ -253,27 +256,27 @@ func (s *Definitions) lookupVPE(name string) (Resource, error) {
...
 		}
-		return Resource{name, ips, ResourceTypeSubnet}, nil
+		return Resource{name, cidrs, ResourceTypeSubnet}, nil
 	}
 	return Resource{}, containerNotFoundError(name, ResourceTypeSegment)
 }
 
 func (s *Definitions) lookupCidrSegment(name string) (Resource, error) {
-	ips := []IP{}
+	cidrs := []*ipblock.IPBlock{}
```
Comment (zivnevo) [inline pkg/ir/spec.go:274]: preallocate memory

File: pkg/synth/acl.go [hunk 3]
```diff
@@ -124,15 +127,15 @@ func resourceRelevantToACL(e ir.ResourceType) bool {
...
-func resourcesContainedInCidr(s *ir.Spec, resourceIP ir.IP, resource ir.Resource) []ir.IP {
+func subnetsContainedInCidr(s *ir.Spec, cidr *ipblock.IPBlock, resource ir.Resource) []*ipblock.IPBlock {
 	if resource.Type != ir.ResourceTypeCidr {
-		return []ir.IP{resourceIP}
+		return []*ipblock.IPBlock{cidr}
 	}
 	cidrSegmentDetails := s.Defs.CidrSegments[resource.Name]
-	cidrDetails := cidrSegmentDetails.Cidrs[ir.CidrFromIP(resourceIP)]
-	retVal := make([]ir.IP, len(cidrDetails.ContainedSubnets))
+	cidrDetails := cidrSegmentDetails.Cidrs[cidr]
```
Comment (zivnevo) [inline pkg/synth/acl.go:135]: Are these two lookups guaranteed to be successful?

File: pkg/io/jsonio/unmarshal.go [hunk 4]
```diff
@@ -350,20 +356,14 @@ func unmarshal(filename string) (*spec.Spec, error) {
 	return jsonSpec, err
 }
 
-func parseOverlappingVpcs(cidr ipblock.IPBlock, vpcs map[ir.ID]*ir.VPCDetails) ([]ir.ID, error) {
+func parseOverlappingVpcs(cidr ipblock.IPBlock, vpcs map[ir.ID]*ir.VPCDetails) []ir.ID {
```
Comment (zivnevo) [inline pkg/io/jsonio/unmarshal.go:359]:
```suggestion
func parseOverlappingVpcs(cidr *ipblock.IPBlock, vpcs map[ir.ID]*ir.VPCDetails) []ir.ID {
```
Better stick to the convention of passing pointers to IPBlocks

File: pkg/io/confio/read_defs.go [hunk 5]
```diff
@@ -161,20 +190,8 @@ func validateVpcs(vpcs map[ir.ID]*ir.VPCDetails) error {
...
-					address2, err := ipblock.FromCidr(addressPrefix2.String())
-					if err != nil {
-						return err
-					}
-					if !address1.Intersect(address2).IsEmpty() {
-						return fmt.Errorf("vpcs %s and %s are overlapping", vpcName1, vpcName2)
-					}
-				}
+			if !vpcDetails1.AddressPrefixes.Intersect(vpcDetails2.AddressPrefixes).IsEmpty() {
+				return fmt.Errorf("vpcs %s and %s are overlapping", vpcName1, vpcName2)
```
Comment (zivnevo) [inline pkg/io/confio/read_defs.go:194]:
```suggestion
				return fmt.Errorf("vpcs %s and %s have overlapping IP address spaces", vpcName1, vpcName2)
```

**Review by zivnevo — CHANGES_REQUESTED** [review]
Overall, looks great. Just a few small things

File: test/data/acl_cidr_segments1/conn_spec.json [hunk 6]
```diff
@@ -3,7 +3,7 @@
         "cidrSegment": {
             "type": "cidr",
             "items": [
-                "10.240.3.0/23"
+                "10.240.2.0/23"
```
[the remark this answers is not in the document]
Comment (YairSlobodin1) [reply test/data/acl_cidr_segments1/conn_spec.json:6]: The 2 and 3 differ in 24th bit of the IP address.

File: pkg/synth/acl.go (hunk 3 above)
Comment (YairSlobodin1) [reply pkg/synth/acl.go:135]: yes, it is checked in `unmarshal` function.

**Review by zivnevo — APPROVED** [review]
LGTM

**Pull request merged**